### ISSUE TITLE
need more division by zero check

### ISSUE THREAD

**Issue by SleepyRoy** [vlang/v#4629]
**V version:** 
latest master with mingw64

**OS:** 
win10

**What did you do?**

case 1: zero stored in variable
```V
fn main() {
	a := 0
	println(1/a)
}
```

case 2: zero other than `int`(i32) type
```V
fn main() {
	println(1/i64(0))
}
```

**What did you expect to see?**
division by zero error
 
**What did you see instead?**
code compiled and exe crashed

**Comment by yuyi98** [comment]
Yes, there are a lot of situations.

**Comment by SleepyRoy** [comment]
@yuyi98 The key is to check whether the value (not just literal) is zero, but I don't know how to do it. 😂

**Comment by medvednikov** [comment]
there's `if is IntegerLital`
need to also do `if is FloatLiteral`

**Comment by SleepyRoy** [comment]
@medvednikov 
Thanks. But how to check value is zero? For example `infix_expr.right.str().int() == 0` only works for literal `0`, not `i64(0)` or 0 stored in variable.

**Comment by yuyi98** [comment]
I think that current processing mechanism cannot be calculated at the check stage and can only rely on literal judgement.

**Comment by Delta456** [comment]
@SleepyRoy Something like this

```v
if infix_expr.right.str().int() == 0 || infix_expr.right.str().i8() == 0 || infix_expr.right.str().u8() == 0 || infix_expr.right.str().u16() == 0 || more casting checks...
```

**Comment by medvednikov** [comment]
@Delta456 really?

`nfix_expr.right.str() == '0'`

**Comment by SleepyRoy** [comment]
@yuyi98 Agreed. And if so, detecting `i64(0)` is OK because at least `0` is in it, but detecting 0 stored in variable would really be a problem - You only get the variable name.

**Comment by Delta456** [comment]
@medvednikov oh nvm , should had looked at more clearly

**Comment by SleepyRoy** [comment]
@medvednikov @Delta456 
Thanks for suggestions. I just did some experiments and basically like @yuyi98 said, `infix_expr.right.str()` only let you obtain the literal value, not value behind.
That's to say, detecting `1/0` is OK because `infix_expr.right.str() == '0'` is true, but detecting `1/i64(0)` or `1/foo` (foo stores 0) is a problem - `infix_expr.right.str()` is `i64(0)` and `foo` respectively for the two cases.
